portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -16,33 +16,56 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletException;
  
 /** Servlet that adds comments to datastore */
 @WebServlet("/data") 
 public class DataServlet extends HttpServlet 
 { 
+  public void init() throws ServletException {
+    try {
+        languageService = LanguageServiceClient.create();
+    } catch (Exception e) {
+        throw new ServletException("I/O exception thrown", e);
+    }
+  }
+  
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     String name = request.getParameter("user-name");
     String message = request.getParameter("user-comment");
     long timestamp = System.currentTimeMillis();
 
-    if(name.length() != 0 && message.length() != 0){
+    if (name.length() != 0 && message.length() != 0) {
         Entity taskEntity = new Entity("Comment");
+
         taskEntity.setProperty("name", name);
         taskEntity.setProperty("message", message);
+        taskEntity.setProperty("score", getSentimentScore(message));
         taskEntity.setProperty("timestamp", timestamp);
         datastore.put(taskEntity);
     }
 
     response.sendRedirect("/index.html#comments-panel");
   }
+
+  public double getSentimentScore(String message) throws IOException {
+    Document doc = Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    double score = sentiment.getScore();
+    languageService.close();
```

Comment:
Should this method be responsible for calling `close()`? What if someone accidentally calls `getSentimentScore()` twice in this handler?

Reply:
I'll close the client in the Servlet's destroy method